src/test/scala/firrtlTests/AnnotationTests.scala

```diff
@@ -0,0 +1,57 @@
+package firrtlTests
+
+import java.io.StringWriter
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+
+import firrtl.{Parser,Circuit}
+import firrtl.{
+   Named,
+   ModuleName,
+   ComponentName,
+   CircuitAnnotation,
+   StringAnnotation,
+   BrittleCircuitAnnotation,
+   UnknownCAKind,
+   Compiler,
+   CompilerResult,
+   Annotation,
+   RenameMap,
+   VerilogCompiler
+}
+
+/**
+ * An example methodology for testing Firrtl annotations.
+ */
+abstract class AnnotationSpec extends FlatSpec {
+   def parse (s: String): Circuit = Parser.parse("",s.split("\n").toIterator)
+   def compiler: Compiler
+   def input: String
+   def getResult (annotation: CircuitAnnotation): CompilerResult = {
+      val writer = new StringWriter()
+      compiler.compile(parse(input), Seq(annotation), writer)
+   }
+}
```

Comment:
Yeah, I'm having a hard time figuring out the best parent class for all the different types of tests we want to run. I'm content having multiple parent classes, and once we settle more on our testing strategy, we can look for what is in common and merge them later.